tagsets/query_modules.php

```diff
@@ -128,6 +128,10 @@ function query__get_query_form_prototypes($hide_modules=array(),$experiment_id="
                     $form_query_fields[]=$tfield;
                 }
         }
+        $int_fields=participant__get_internal_freetext_search_fields();
+        foreach ($int_fields as $ifield) {
```

Comment:
Technically you do not need `foreach` here, line below will do the same:
`$form_query_fields=array_values($int_fields);`

Reply:
Note that $form_query_fields may be pre-filled with some values in the lines above that. The code line you suggest would undo that and create a new $form_query_fields array only filled with the new values. There seems to be no array_push() function that adds values from an array (rather than some different variables), and array_merge() would retain (ans possible overwrite) the keys.